runtime/src/main/java/org/corfudb/runtime/collections/ProtobufIndexer.java

```diff
@@ -38,28 +38,45 @@ public class ProtobufIndexer implements Index.Registry<Message, CorfuRecord<Mess
         registerSecondaryIndex(payloadSchema, schemaOptions);
     }
 
-    private <T> Index.Spec<Message, CorfuRecord<Message, Message>, ?>
-    getIndex(String indexPath, String indexName, FieldDescriptor fieldDescriptor) {
-        return new Index.Spec<>(
-                () -> indexPath,
-                () -> indexName,
-                (Index.Function<Message, CorfuRecord<Message, Message>, T>)
-                        (key, val) -> ClassUtils.cast(val.getPayload().getField(fieldDescriptor)));
+    static class IndexFieldMapper {
+
+        // Use an int array instead of a collection (i.e., Map to eliminate auto-boxing and related garbage)
+        private final int[] indexMap;
+
+        private final int unset = -2;
```

Comment:
just curious, why -2 and not -1?

Reply:
It's just an invalid index, -2 has no significance.